### ISSUE TITLE
unable to build picard

### ISSUE THREAD

**Issue by zo0z** [broadinstitute/picard#624]
Hi!

I'm having some trouble installing picard on my server. I cloned the repo and verified that there is a hidden .git directory as per the documentation. However, it fails with errors:

```
$ ./gradlew shadowJar
FAILURE: Build failed with an exception.
* Where:
Build file '/usr/src/picard-tools/picard-tools-2.4.1/build.gradle' line: 56
* What went wrong:
A problem occurred evaluating root project 'picard'.
> Cannot invoke method getURLs() on null object
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
BUILD FAILED
Total time: 5.344 secs

```

I got the same error log when ./gradlew compileJava, and the complete log is below:

```
* Exception is:
org.gradle.api.GradleScriptException: A problem occurred evaluating root project 'picard'.
	at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:93)
	at org.gradle.configuration.DefaultScriptPluginFactory$ScriptPluginImpl$2.run(DefaultScriptPluginFactory.java:157)
	at org.gradle.configuration.ProjectScriptTarget.addConfiguration(ProjectScriptTarget.java:72)

	at org.gradle.configuration.DefaultScriptPluginFactory$ScriptPluginImpl.apply(DefaultScriptPluginFactory.java:162)
	at org.gradle.configuration.project.BuildScriptProcessor.execute(BuildScriptProcessor.java:38)
	at org.gradle.configuration.project.BuildScriptProcessor.execute(BuildScriptProcessor.java:25)
	at org.gradle.configuration.project.ConfigureActionsProjectEvaluator.evaluate(ConfigureActionsProjectEvaluator.java:34)
	at org.gradle.configuration.project.LifecycleProjectEvaluator.evaluate(LifecycleProjectEvaluator.java:55)
	at org.gradle.api.internal.project.AbstractProject.evaluate(AbstractProject.java:529)
	at org.gradle.api.internal.project.AbstractProject.evaluate(AbstractProject.java:90)
	at org.gradle.execution.TaskPathProjectEvaluator.configureHierarchy(TaskPathProjectEvaluator.java:42)
	at org.gradle.configuration.DefaultBuildConfigurer.configure(DefaultBuildConfigurer.java:35)
	at org.gradle.initialization.DefaultGradleLauncher$2.run(DefaultGradleLauncher.java:125)
	at org.gradle.internal.Factories$1.create(Factories.java:22)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:52)
	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:122)
	at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:32)
	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:99)
	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:93)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:62)
	at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:93)
	at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:82)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:94)
	at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:43)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:28)
	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:81)
	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:46)
	at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:51)
	at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:28)
	at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:43)
	at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:173)
	at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:241)
	at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:214)
	at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:35)
	at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:24)
	at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:207)
	at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:169)
	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
	at org.gradle.launcher.Main.doAction(Main.java:33)
	at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
	at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:55)
	at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:36)
	at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
	at org.gradle.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:30)
	at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:129)
	at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
Caused by: java.lang.NullPointerException: Cannot invoke method getURLs() on null object
	at build_4eijkve03soqb18d637dc98wz$_run_closure4.doCall(/usr/src/picard-tools/picard-tools-2.4.1/build.gradle:56)
	at org.gradle.api.internal.ClosureBackedAction.execute(ClosureBackedAction.java:67)
	at org.gradle.util.ConfigureUtil.configure(ConfigureUtil.java:130)
	at org.gradle.util.ConfigureUtil.configure(ConfigureUtil.java:91)
	at org.gradle.api.internal.project.AbstractProject.dependencies(AbstractProject.java:892)
	at org.gradle.api.internal.BeanDynamicObject$MetaClassAdapter.invokeMethod(BeanDynamicObject.java:236)
	at org.gradle.api.internal.BeanDynamicObject.invokeMethod(BeanDynamicObject.java:130)
	at org.gradle.api.internal.CompositeDynamicObject.invokeMethod(CompositeDynamicObject.java:150)
	at org.gradle.groovy.scripts.BasicScript.methodMissing(BasicScript.java:79)
	at build_4eijkve03soqb18d637dc98wz.run(/usr/src/picard-tools/picard-tools-2.4.1/build.gradle:52)
	at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:91)
	... 50 more


BUILD FAILED

Total time: 5.411 secs

```

Other session details:

```
$ git --version
git version 1.9.1
$ javac -version
javac 1.8.0_91
```

Thank you in advance,
Azza

**Comment by lbergelson** [comment]
Hi Azza,
Thanks for the detailed report, and sorry you're running into problems with gradle build. I'm looking into this one but I'm not yet able to reproduce it myself.

It looks like for some reason you're getting a null when `getSystemToolClassLoader()` is being called at line 56 in `build.gradle`.  

```
    compile(files(((URLClassLoader) ToolProvider.getSystemToolClassLoader()).getURLs()))
```
The javadoc for getSystemToolClassLoader() says that null is a valid option, so it's a case we should be handling. That said, I've never seen it happen on any of our systems, so it must be some difference in your java installation.  

A few questions:

1. What OS are you running?
2. Are you sure you have a complete [JDK](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html) instead of a JRE (since you have javac I would have assume you have a JDK, but maybe it comes with the JRE now too)
3. Are you running oracle java or a different jvm? (again, it looks like you have an oracle installation, but always good to check for sure)

For potential solutions, could you try replacing line 56 in build.gradle with the following and rerunning:
```
    compile files("${System.properties['java.home']}/../lib/tools.jar")
```
If that doesn't work, can you check that you have a `tools.jar` file in your java installation?  

As a workaround until we can solve the problem building it, you could use the [prebuilt jar](https://github.com/broadinstitute/picard/releases/download/2.6.0/picard.jar)

**Comment by zo0z** [comment]
Hi!

Thank you very much for your prompt response, much appreciated.. 

Here are some answers:

1. I'm running on Ubuntu 14.04

1. Yes, it is a complete jdk as far as I can tell. Here are some outputs:
```
java -version
java version "1.8.0_91"
Java(TM) SE Runtime Environment (build 1.8.0_91-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.91-b14, mixed mode)
```

1. To be honest, I downloaded the jdk from an alternative repo than Oracle, because the Oracle website and resources are banned in my country, in compliance with US laws (or so the note suggests)... However, the file itself has the same md5sum as the Oracle website, and also the same name and version; and my understanding is that it is just a mirror...

I didn't change the file in build.gradle, but my problem seems to have been resolved. Basically, I recloned the repo to a different directory (into my home), and it basically worked... The build folder and subdirecotries have been created and therefore I'm concluding that it was a permission issue all along (I was trying to install in /usr/src/, and using sudo with all commands was not sufficient to create the various build files)..

Apologies for the inconvenience, and thank you again..
Azza
